portal/src/main/webapp/js/src/dashboard/iviz.js

```diff
@@ -8178,3 +8179,191 @@ window.LogRankTest = (function(jStat) {
+            'stats', 'updateStats', 'showShareButton'
+        ],
+        data: function() {
+            return {
+                savedVC: null
+            };
+        }, methods: {
+            saveCohort: function() {
+                var _self = this;
+                _self.updateStats = true;
+                _self.$nextTick(function() {
+                    _self.addNewVC = true;
+                });
+            }
+        },
+        ready: function() {
+            var self_ = this;
+            var previous_selectedCases = {};
+            if (this.showShareButton) {
+                $('.share-virtual-study').qtip({
+                    style: {
+                        classes: 'qtip-light qtip-rounded qtip-shadow'
+                    },
+                    show: {event: 'mouseover', ready: false},
+                    hide: {fixed: true, delay: 200, event: 'mouseleave'},
+                    position: {
+                        my: 'bottom center',
+                        at: 'top center',
+                        viewport: $(window)
+                    },
+                    content: 'Share Virtual Study'
+                });
+                $('.share-cohort-btn').qtip({
+                    style: {
+                        classes: 'qtip-light qtip-rounded qtip-shadow ' +
+                        'iviz-share-cohort-btn-qtip'
+                    },
+                    show: {event: 'click', ready: false},
+                    hide: false,
+                    position: {
+                        my: 'top center',
+                        at: 'bottom center',
+                        viewport: $(window)
+                    },
+                    events: {
+                        render: function(event, api) {
+                            var tooltip = $('.iviz-share-cohort-btn-qtip .qtip-content');
+                            
+                            tooltip.find('.share-cohort').click(function() {
+                                tooltip.find('.shared').css('display', 'none');
+                                tooltip.find('.dialog').css('display', 'none');
+                                api.reposition();
+                                
+                                // Copy virtual study link to clipboard
+                                var $temp = $("<input>");
+                                $("body").append($temp);
+                                $temp.val(tooltip.find('.virtual-study-link').attr('href')).select();
+                                document.execCommand("copy");
+                                // Check if users copy url successfully
+                                if ($temp.val() === tooltip.find('.virtual-study-link').attr('href')) {
+                                    tooltip.find('.shared').css('display', 'block');
+                                    tooltip.find('.dialog').css('display', 'none');
+                                    $temp.remove();
```

Comment:
I think you should remove `$temp` either way, right? (outside the `if` statement)

Reply:
Yes, I should move it outside the `if` statement.